ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java

```diff
@@ -151,7 +151,11 @@ static ShadingLanguageVersion webGlFromVersionString(String versionString) {
 
   boolean isWebGl();
 
-  boolean restrictedArrayIndexing();
+  /**
+   * GLSL versions 1.2+ and ESSL versions 3.0+ support array constructors.
+   * @return true if and only if array constructors are supported.
+   */
+  boolean supportedArrayConstructors();
```

Comment:
Should we add some very simple tests?

Reply:
I suppose the new shaders act as tests

Reply:
Hmm, I suppose we could have some tests that go through the claimed properties supported by shading languages and generate a simple shader that has the property, checking that it is valid if and only if we deem the property supported.

I'll open an issue for that, rather than it being something we do in this PR (as it is relevant to all the other properties too.)

Reply:
Opened #882 for this - and will merge this PR now.